### ISSUE TITLE
datadog_synthetics_test resource does not apply options_list.ignore_server_certificate_error

### ISSUE THREAD

**Issue by thomasspatzier** [DataDog/terraform-provider-datadog#1692]
### Terraform Version

Terraform v1.3.5
on linux_arm64
+ provider registry.terraform.io/datadog/datadog v3.19.1
+ provider registry.terraform.io/hashicorp/azuread v2.31.0
+ provider registry.terraform.io/hashicorp/azurerm v3.37.0
+ provider registry.terraform.io/hashicorp/random v3.4.3

### Affected Resource(s)

-   datadog_synthetics_test

### Terraform Configuration Files

```hcl
resource "datadog_synthetics_test" "my_api_test" {
  name   = "My API Test"
  status = "live"

  type    = "api"
  subtype = "http"

  request_definition {
    method = "GET"
    url    = "https://my-api.example.com"
  }

  assertion {
    type     = "statusCode"
    operator = "is"
    target   = "200"
  }

  locations = [ "aws:eu-central-1" ]

  options_list {
    tick_every                      = 600
    ignore_server_certificate_error = true
    follow_redirects                = true

    retry {
      count    = 3
      interval = 3000
    }
  }

  request_client_certificate {
    cert {
      content = file("cert.pem")
    }
    key {
      content = file("keydec.pem")
    }
  }
}
```

### Expected Behavior

The "Ignore server certificate error" option should be checked when I inspect the synthetic test in the Datadog portal and tests should work with self-signed certs.

### Actual Behavior

The "Ignore server certificate error" option is not checked when I inspect the test in the Datadog portal and tests fail because of self-signed certs.

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:

1. `terraform apply`

**Comment by ilya-zhorov** [comment]
facing similar with browser_tests

```
  options_list {
    tick_every = 900

    ignore_server_certificate_error = true

    retry {
      count    = 0      # Number of retries
    }

    monitor_options {
        renotify_interval = 0
    }
  }
```

facing the error

```
Error: Unsupported argument
│
│   on browsers.tf line 111, in resource "datadog_synthetics_test" "test":
│  161:     ignore_server_certificate_error = true
│
│ An argument named "ignore_server_certificate_error" is not expected here.
```

Importing already created resource with UI flag in Advanced options for ignore-server-certificate-error - ignored. 

https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/synthetics_test#ignore_server_certificate_error

**Comment by nkzou** [comment]
From some testing, it looks like the `ignore_server_certificates_error` field is only respected when the synthetics test is a browser test.
When working with a synthetics API test, the `allow_insecure` field toggles the correct "Ignore server certificates error" option in the web UI.

This is pretty confusing and not well documented, we'll look into improving this. Feel free to reopen this issue if the `allow_insecure` field doesn't solve your issue.

**Comment by thomasspatzier** [comment]
Thanks for the clarification. I just tried this out with `type = "browser"` and it works in my case. It was just confusing to see the `ignore_server_certificate_error` option in the UI and being able to set it via the UI for a test of `type = "api"`.

BTW, I tried using the `allow_insecure` field in combination with the `ignore_server_certificates_error` and think it didn't work. But when I use just `allow_insecure`, it seems to work.

So yes, some updates to the docs would be helpful. But thanks for the help!
